### PR TITLE
Generate unique index for PartitionColumns

### PR DESCRIPTION
**Pull request by bhhari**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java [hunk 1]
```diff
@@ -352,6 +352,9 @@ public void testArrays()
     {
         Random random = new Random(0);
 
+        tester.testRoundTrip(arrayType(VARCHAR),
+                ImmutableList.of(ImmutableList.of()), IS_NOT_NULL);
```
Comment (highker) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java]: put these two lines into one

Comment (bhhari) [reply presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java]: actually i don't need this test, this was a temporary change for a different issue, will remove it

**Comment by bhhari** [comment]
@highker , fixed the comment, can you also confirm that dynamically assigning the partition column index does not effect existing batch reader code. Its good to merge otherwise. Thanks

**Comment by highker** [comment]
@bhhari, it looks to me it won't but I'm not that confident. Could you this patch running on verifier against prod?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 2]
```diff
@@ -857,12 +858,13 @@ public static SchemaTableName schemaTableName(ConnectorTableHandle tableHandle)
...
 
         List<Column> partitionKeys = table.getPartitionColumns();
+        int partitionColumnIndex = MAX_PARTITION_KEY_COLUMN_INDEX;
         for (Column field : partitionKeys) {
             HiveType hiveType = field.getType();
             if (!hiveType.isSupportedType()) {
                 throw new PrestoException(NOT_SUPPORTED, format("Unsupported Hive type %s found in partition keys of table %s.%s", hiveType, table.getDatabaseName(), table.getTableName()));
             }
-            columns.add(new HiveColumnHandle(field.getName(), hiveType, hiveType.getTypeSignature(), -1, PARTITION_KEY, field.getComment()));
+            columns.add(new HiveColumnHandle(field.getName(), hiveType, hiveType.getTypeSignature(), partitionColumnIndex--, PARTITION_KEY, field.getComment()));
```
Comment (wenleix) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: nit: We don't generally use `++`/`--` in Presto codebase. I would just do

```
columns.add(new HiveColumnHandle(field.getName(), hiveType, hiveType.getTypeSignature(), partitionColumnIndex, PARTITION_KEY, field.getComment()));
partitionColumnIndex--;
```

Comment (wenleix) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: Never mind, I see the original code use `--`

**Comment by highker** [comment]
@bhhari, let us know once you have the verifier run.

**Comment by bhhari** [comment]
fixes #13522

**Pull request merged**